src/main/java/lesson9Refactoring1/Employee.java

```diff
@@ -0,0 +1,60 @@
+package lesson9Refactoring1;
+
+import java.math.BigDecimal;
+
+public class Employee {
+
+    public static final String WHITE_SPACE = " ";
+    private final String name;
+    private final String surname;
+    private int age;
+    private BigDecimal baseSalary;
+    private BigDecimal bonus;
+
+    public Employee(String name, String surname) {
+        this.name = name;
+        this.surname = surname;
+    }
+
+    public Employee(String name, String surname, int age) {
+        this.name = name;
+        this.age = age;
+        this.surname = surname;
+    }
+
+    public Employee(String name, String surname, int age, BigDecimal baseSalary) {
+        this.name = name;
+        this.age = age;
+        this.baseSalary = baseSalary;
+        this.surname = surname;
+    }
+
+    public Employee(String name, int age, BigDecimal baseSalary, BigDecimal bonus, String surname) {
```

Comment:
Zamiast w każdym z konstruktorów powtarzać this.x = x; itp. można po prostu użyć jednego konstruktora w drugim :)
Będą to tak zwane wywołania teleskopowe.

Reply:
zaaplikowano :)